apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/define/KafkaProducerInstrumentation.java

```diff
@@ -46,22 +46,35 @@
     public static final String INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.kafka.KafkaProducerInterceptor";
     public static final String ENHANCE_CLASS = "org.apache.kafka.clients.producer.KafkaProducer";
     public static final String ENHANCE_METHOD = "doSend";
-    public static final String CONSTRUCTOR_INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.kafka.ProducerConstructorInterceptor";
-    public static final String CONSTRUCTOR_INTERCEPTOR_FLAG = "org.apache.kafka.clients.producer.ProducerConfig";
-    public static final String METHOD_FLAG = "org.apache.kafka.clients.producer.Callback";
+    public static final String CONSTRUCTOR_INTERCEPTOR_CLASS1 = "org.apache.skywalking.apm.plugin.kafka.ProducerConstructorInterceptor";
```

Comment:
I'm not sure if I can divide it into two instrumentations to do it through witness class. Because it dose not look graceful enough. Coud you try it?

Reply:
I think two instrumentation definitions are better and recommended.

Reply:
At first I tried to do this by witness classe, but I didn't find any different classes in 2.1.0 and 2.1.1...

Reply:
Or I can use two instrumentation definitions enhance two constructors, only hitting one at a time

Reply:
@stalary I think @dmsolr is just asking for separating codes into two classes. Not about the instrumentation logic. Make sense?

Reply:
Seems to separate code I need to implement two duplicate getInstanceMethodsInterceptPoints to enhance doSend.

Reply:
Yes, to enhance CONSTRUCTOR_INTERCEPTOR_CLASS and CONSTRUCTOR_INTERCEPTOR_CLASS2.

Reply:
I see. I'll do it

Reply:
@dmsolr I have modified it. Please check it again

Reply:
That looks good.